Refactored scraping app and added tests

kimjudyh opened the pull request:
Created different classes for resort scrapers, making it more modular and easier to add/remove resorts to be scraped.

kimjudyh commented:
Website is in a usable state to check conditions and twitter feeds:
https://ski-conditions-dev.herokuapp.com/conditions/
https://ski-conditions-dev.herokuapp.com/conditions/twitter

clintonb commented on Pipfile:
```diff
@@ -15,6 +15,7 @@ gevent = "~=1.4"
 gunicorn = "~=19.9"
 psycopg2-binary = "~=2.8"
 requests = "~=2.22.0"
+responses = "~=0.10.7"
```
This is only needed for development.

clintonb commented on scraping/heavenly_scraper.py:
```diff
@@ -0,0 +1,55 @@
+def ski_scraper(url):
```
This entire directory should be removed.

kimjudyh replied:
Removed directory.

clintonb commented on ski_conditions/apps/app_scraping/migrations/0001_initial.py:
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 2.2.6 on 2019-11-21 08:57
```
Never modify a migration once it has been merged and deployed. There are exceptions to this rule, but for now you should always create new migrations.

clintonb commented on ski_conditions/apps/app_scraping/views.py:
```diff
@@ -9,5 +9,22 @@ def index(request):
     return HttpResponse(template.render(context, request))
+
+
+def twitter(request):
```
What’s the purpose of this view? Why not use a template?

kimjudyh replied:
Moved twitter feed embedding html into a template.

clintonb commented on ski_conditions/apps/app_scraping/tests/test_commands.py:
```diff
@@ -0,0 +1,125 @@
+@pytest.mark.django_db()
+def test_snow_report():
+    '''
+    test snow report scraping
```
This comment doesn’t tell us anything we don’t already know from the method name. It can be removed.

kimjudyh replied:
Removed comment

clintonb commented on ski_conditions/apps/app_scraping/tests/test_commands.py:
```diff
@@ -0,0 +1,125 @@
+    '''
+
+    # use saved html instead of accessing live site
+    body_Kirkwood = open(os.path.join(os.path.dirname(__file__), 'data', 'kirkwood.html'))
```
DRY: don’t repeat yourself. Use a loop to clean up the duplication:

```python
data = (
  (KirkwoodScraper.url, 'kirkwood.html',),
)

for datum one data:
  body = ... # load from disk
  responses.add(...)
```

kimjudyh replied:
Removed duplication in commit 4274957

clintonb commented on ski_conditions/apps/app_scraping/tests/test_commands.py:
```diff
@@ -0,0 +1,125 @@
+@pytest.mark.django_db()
+def test_scraping():
+    '''
+    test lift and trail number scraping
```
Similar comments as below. This comment is not useful. Also, how does this test differ from the one below?

DRY it up.

kimjudyh replied:
Merged tests back into one. I separated them because of a pylint issue that said there were too many statements. But after replacing the duplicate statements with a loop, that issue was resolved.

kimjudyh commented:
@clintonb I've addressed most of the changes requested and have also updated the homepage to have a description and links to the other pages. I think it's ready to merge!
